after/syntax/sml.vim

```diff
@@ -13,60 +13,69 @@
 "
 "       au! FileType sml setlocal conceallevel=2
 "
-
 " Fix wonky highlighting for => operator
 syn match smlOperator  "=>"
 
 " Also highlight 'NOTE' and 'Note' in comments
 syn keyword  smlTodo contained NOTE Note
 
-" Highlight type variables (i.e., tokens that look like 'xyz)
-syn match smlType "\<'\w\+\>"
-
-" Add conceal characters for common greek letters
-" that are used for type variables
-syn match smlType "'a\>"               conceal cchar=α
-syn match smlType "'alpha\>"           conceal cchar=α
-syn match smlType "'b\>"               conceal cchar=β
-syn match smlType "'beta\>"            conceal cchar=β
-syn match smlType "'c\>"               conceal cchar=γ
-syn match smlType "'gamma\>"           conceal cchar=γ
-syn match smlType "'d\>"               conceal cchar=δ
-syn match smlType "'delta\>"           conceal cchar=δ
-syn match smlType "'e\>"               conceal cchar=ε
-syn match smlType "'epsilon\>"         conceal cchar=ε
-
-syn match smlType "'zeta\>"            conceal cchar=ζ
-syn match smlType "'eta\>"             conceal cchar=η
-syn match smlType "'theta\>"           conceal cchar=θ
-syn match smlType "'kappa\>"           conceal cchar=κ
-syn match smlType "'lambda\>"          conceal cchar=λ
-
-syn match smlType "'m\>"               conceal cchar=μ
-syn match smlType "'mu\>"              conceal cchar=μ
-syn match smlType "'n\>"               conceal cchar=ν
-syn match smlType "'nu\>"              conceal cchar=ν
-
-syn match smlType "'xi\>"              conceal cchar=ξ
-
-syn match smlType "'p\>"               conceal cchar=π
-syn match smlType "'pi\>"              conceal cchar=π
-syn match smlType "'r\>"               conceal cchar=ρ
-syn match smlType "'rho\>"             conceal cchar=ρ
-syn match smlType "'s\>"               conceal cchar=σ
-syn match smlType "'sigma\>"           conceal cchar=σ
-syn match smlType "'t\>"               conceal cchar=τ
-syn match smlType "'tau\>"             conceal cchar=τ
-
-syn match smlType "'upsilon\>"         conceal cchar=υ
-syn match smlType "'phi\>"             conceal cchar=ϕ
-syn match smlType "'x\>"               conceal cchar=χ
-syn match smlType "'chi\>"             conceal cchar=χ
-
-syn match smlType "'psi\>"             conceal cchar=ψ
-
-syn match smlType "'w\>"               conceal cchar=ω
-syn match smlType "'omega\>"           conceal cchar=ω
+" Add conceal characters for common Greek letters
+" that are used for type variables.
+function! s:ConcealTyvar(tyvar, cchar)
+    let l:tick = "'"
+    if exists('g:sml_greek_tyvar_show_tick') && g:sml_greek_tyvar_show_tick
+        let l:tick = '\%(' . l:tick . '\)\@<='
```

Comment:
I'm new to this `\%(...\)` construct. What's the benefit? `:help` tells me this:

```
\%(\)   A pattern enclosed by escaped parentheses.
        Just like \(\), but without counting it as a sub-expression.  This
        allows using more groups and it's a little bit faster.
```